app/utils/categories.test.ts

```diff
@@ -0,0 +1,663 @@
+                isUnread: false,
+            } as MyChannelModel,
+        }, {
+            ...data[1],
+            myChannel: {
+                id: data[1].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }];
+
+        result = filterAutoclosedDMs('direct_messages', 3, 'me', '', mockRecent1, [], {});
+        expect(result[0].channel.id).toEqual('dm1');
+
+        const mockRecent2: ChannelWithMyChannel[] = [{
+            ...data[0],
+            myChannel: {
+                id: data[0].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }, {
+            ...data[1],
+            myChannel: {
+                id: data[1].myChannel.id,
+                lastViewedAt: 2,
+                isUnread: false,
+            } as MyChannelModel,
+        }];
+
+        result = filterAutoclosedDMs('direct_messages', 3, 'me', '', mockRecent2, [], {});
+        expect(result[0].channel.id).toEqual('dm2');
+
+        const mockSameLastViewed: ChannelWithMyChannel[] = [{
+            ...data[0],
+            myChannel: {
+                id: data[0].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }, {
+            ...data[1],
+            myChannel: {
+                id: data[1].myChannel.id,
+                lastViewedAt: 1,
+                isUnread: false,
+            } as MyChannelModel,
+        }];
+
+        result = filterAutoclosedDMs('direct_messages', 3, 'me', '', mockSameLastViewed, [], {});
+        expect(result[0].channel.id).toEqual('dm1');
+    });
+
+    test('filterManuallyClosedDms', async () => {
+        const db = DatabaseManager.getServerDatabaseAndOperator(serverUrl);
+
+        const closedDMsPrefs = [{
+            id: 'pref1',
+            userId: 'me',
+            category: Preferences.CATEGORIES.DIRECT_CHANNEL_SHOW,
+            name: 'dm1',
+            value: 'true',
+        }, {
+            id: 'pref2',
+            userId: 'me',
+            category: Preferences.CATEGORIES.DIRECT_CHANNEL_SHOW,
+            name: 'dm2',
+            value: 'true',
+        }, {
+            id: 'pref3',
+            userId: 'me',
+            category: Preferences.CATEGORIES.GROUP_CHANNEL_SHOW,
+            name: 'gm1',
+            value: 'false',
+        }] as PreferenceModel[];
+
+        const {channels, myChannels} = buildChannelsData();
+        const data = await createChannelsDbRecords(db, channels, myChannels);
+
+        const notifyProps = myChannels.reduce<Record<string, Partial<ChannelNotifyProps>>>((obj, m) => {
+            obj[m.channel_id] = m.notify_props;
+            return obj;
+        }, {});
+
+        const result = filterManuallyClosedDms(data, notifyProps, closedDMsPrefs, 'me');
+        expect(result.length).toEqual(3);
+    });
+
+    test('sortChannels', async () => {
```

Comment:
0/5 we could test to make sure we handle the locale. I got this from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare

```
console.log("ä".localeCompare("z", "de")); // a negative value: in German, ä sorts before z
console.log("ä".localeCompare("z", "sv")); // a positive value: in Swedish, ä sorts after z
```